chain/chain/src/chain.rs

```diff
@@ -464,20 +565,13 @@ impl Chain {
     }
 
     pub fn save_orphan(&mut self, block: &Block) -> Result<(), Error> {
```

Comment:
For example, you could change function to `save_orphan(&mut self, block: &MaybeValidated<Block>)` and calls 
```block.validate_with(|b|{Chain::check_block_validity(self.runtime_adapter.as_ref(), &self.genesis, b)})``` 
for validation